app-shell-odd/src/config/migrate.ts

```diff
@@ -7,317 +7,71 @@ import {
   OT3_MANIFEST_URL,
 } from '@opentrons/app/src/redux/config'
 
-import type {
-  Config,
-  ConfigV0,
-  ConfigV1,
-  ConfigV2,
-  ConfigV3,
-  ConfigV4,
-  ConfigV5,
-  ConfigV6,
-  ConfigV7,
-  ConfigV8,
-  ConfigV9,
-  ConfigV10,
-  ConfigV11,
-  ConfigV12,
-} from '@opentrons/app/src/redux/config/types'
+import type { Config, ConfigV12 } from '@opentrons/app/src/redux/config/types'
 // format
-// base config v0 defaults
+// base config v12 defaults
 // any default values for later config versions are specified in the migration
 // functions for those version below
 
-export const DEFAULTS_V0: ConfigV0 = {
-  version: 0,
+export const DEFAULTS_V12: ConfigV12 = {
+  version: 12,
   devtools: false,
   reinstallDevtools: false,
-
-  // app update config
-  update: {
-    channel: _PKG_VERSION_.includes('beta') ? 'beta' : 'latest',
-  },
-
-  buildroot: {
-    manifestUrl: OT2_MANIFEST_URL,
-  },
-
-  // logging config
-  log: {
-    level: {
-      file: 'debug',
-      console: 'info',
-    },
-  },
-
-  // ui and browser config
+  update: { channel: _PKG_VERSION_.includes('beta') ? 'beta' : 'latest' },
+  log: { level: { file: 'debug', console: 'info' } },
   ui: {
     width: 1024,
-    height: 768,
-    url: {
-      protocol: 'file:',
-      path: 'ui/index.html',
-    },
-    webPreferences: {
-      webSecurity: true,
-    },
+    height: 600,
+    url: { protocol: 'file:', path: 'ui/index.html' },
+    webPreferences: { webSecurity: true },
+    minWidth: 600,
   },
-
-  // analytics (mixpanel)
   analytics: {
     appId: uuid(),
-    optedIn: true,
-    seenOptIn: false,
+    optedIn: false,
+    seenOptIn: true,
   },
-
-  // user support (intercom)
   support: {
     userId: uuid(),
     createdAt: Math.floor(Date.now() / 1000),
-    name: 'Unknown User',
+    name: null,
     email: null,
   },
-
-  // robot discovery
   discovery: {
     candidates: [],
+    disableCache: false,
   },
-
-  // custom labware files
   labware: {
     directory: path.join(app.getPath('userData'), 'labware'),
+    showLabwareOffsetCodeSnippets: false,
   },
-
-  alerts: {
-    ignored: [],
-  },
-
-  // deprecated fields maintained for safety
+  alerts: { ignored: [] },
   p10WarningSeen: {},
-}
-
-// config version 1 migration and defaults
-const toVersion1 = (prevConfig: ConfigV0): ConfigV1 => {
-  const nextConfig = {
-    ...prevConfig,
-    version: 1 as const,
-    discovery: {
-      ...prevConfig.discovery,
-      disableCache: false,
+  calibration: { useTrashSurfaceForTipCal: null },
+  python: { pathToPythonOverride: null },
+  modules: { heaterShaker: { isAttached: false } },
+  isOnDevice: false,
```

Comment:
```suggestion
  isOnDevice: true,
```
or maybe we can remove this?